server/src/main/java/org/elasticsearch/action/search/SearchRequest.java

```diff
@@ -301,6 +302,24 @@ public ActionRequestValidationException validate() {
             if (source.aggregations() != null) {
                 validationException = source.aggregations().validate(validationException);
             }
+            if (source.rescores() != null) {
+                if (source.from() > 0) {
```

Comment:
We should also check for the case where from is 0 but the window size is smaller than size alone ?
Changing the window size when paginating is also inconsistent so the window size should be thought with this limitation in mind. The initial window size must be greater or equals than the max allowed from+size to avoid pagination issues so I fear that this change will lead to more inconsistency (users that changed their window size on each request while paginating) ?